### PR TITLE
Zappmail: Noisy First Steps

### PR DESCRIPTION
**Pull request by zachmargolis**
. @puls

These changes have Zapp send an email to a hardcoded address every time it finishes a build.

The goal is to have fewer emails, and have them be configurable via GUI, but the priority is to actually get Zapp "talking" first. Watch for more pull requests that address these.

### CODE REVIEW

File: Zapp/ZappRepository.h [hunk 1]
```diff
@@ -17,6 +17,7 @@ extern NSString *const GitFetchSubcommand;
 
 
 @class ZappBuild;
+@class ZappMailConfiguration;
```
Comment (puls) [inline Zapp/ZappRepository.h]: Wait what?

File: Zapp/ZappMessageController.m [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    [combinedHeadersAndMessage writeToFile:temporaryFilePath atomically:YES encoding:NSUTF8StringEncoding error:nil];
+    NSFileHandle *mailFileHandle = [NSFileHandle fileHandleForReadingAtPath:temporaryFilePath];
+    
+    // TODO: get to-address from somewhere
+    NSString *toAddress = @"ios-ci@squareup.com";
+    NSArray *arguments = [NSArray arrayWithObjects:@"-v", toAddress, temporaryFilePath, nil];
+    NSLog(@"running %@", SendmailCommand);
+    
+    // runCommandAndWait must be called from a background queue.
+    dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_LOW, 0), ^{
```
Comment (puls) [inline Zapp/ZappMessageController.m]: You should probably either expose `ZappRepositoryBackgroundQueue` or call `-[ZappRepository runCommand:withArguments:completionBlock:]` with the appropriate additional standard input.

If you call this while you're still on the background queue in `ZappBuild`, of course, it's all moot.

File: Zapp/ZappMessageController.m [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+- (void)sendEmailFromRepository:(ZappRepository*)repository withSubject:(NSString *)subject headers:(NSDictionary *)headers body:(NSString *)body;
+{
+    NSString *subjectHeaderLine = [NSString stringWithFormat:@"Subject: %@", subject];
+    // TODO: break headers into key: value lines
+    NSString *headerLines = @"";
+    NSString *combinedHeadersAndMessage = [[NSArray arrayWithObjects:subjectHeaderLine, headerLines, body, nil] componentsJoinedByString:@"\n"];
+    
+    NSString *temporaryFilePath = [NSString stringWithFormat:@"%@/output.msg", NSTemporaryDirectory()];
+    [combinedHeadersAndMessage writeToFile:temporaryFilePath atomically:YES encoding:NSUTF8StringEncoding error:nil];
+    NSFileHandle *mailFileHandle = [NSFileHandle fileHandleForReadingAtPath:temporaryFilePath];
```
Comment (puls) [inline Zapp/ZappMessageController.m:66]: You don't need a file here, you could use an `NSPipe`. Just sayin'.

File: Zapp/ZappAppDelegate.m [hunk 4]
```diff
@@ -185,6 +186,10 @@
     self.building = YES;
     [build startWithCompletionBlock:^{
         self.building = NO;
+        
+        ZappMessageController *messageController = [ZappMessageController new];
+        [messageController sendMessageForBuild:build];
```
Comment (puls) [inline Zapp/ZappAppDelegate.m]: This should probably just get called straight from `ZappBuild`.

File: Zapp/ZappMessageController.h [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+//  Created by Zach Margolis on 10/5/11.
+//  Copyright (c) 2011 Square, Inc. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+
+@interface ZappMessageController : NSObject
+
+- (void)sendMessageForLatestBuildInRepository:(ZappRepository *)repository;
+- (void)sendMessageForBuild:(ZappBuild *)build;
```
Comment (puls) [inline Zapp/ZappMessageController.h]: Make these class methods? There's no real state on this object.

File: Zapp/ZappRepository.h (hunk 1 above)
Comment (zachmargolis) [inline Zapp/ZappRepository.h]: A previous build had a class to manage mail config state, but I cut that for the first pass. Will remove.

File: Zapp/ZappMessageController.m (hunk 3 above)
Comment (zachmargolis) [inline Zapp/ZappMessageController.m:66]: On it.

Comment (zachmargolis) [inline Zapp/ZappMessageController.m:66]: Scratch that, pipes need to be connected to an output source before writing. That interface is not exposed, so going to expose standardInput as a route with less resistance.

**Pull request merged**